src/lib.rs

```diff
@@ -284,6 +284,33 @@ impl Log for Logger {
     fn flush(&self) {}
 }
 
+/// Prints key values in ": key1=value1, key2=value2" format.
+///
+/// # Notes
+///
+/// Prints ": " itself, only when there is at least one key value pair.
+struct KeyValuePrinter<'a>(&'a dyn kv::Source);
+
+impl<'a> fmt::Display for KeyValuePrinter<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.0.visit(&mut KeyValueVisitor(true, f))
+            .map_err(|_| fmt::Error)
```

Comment:
We should support using the `?` operator here.

Reply:
I think so, the following should work too. But it's the same amount of lines anyway.

```rust
       self.0.visit(&mut KeyValueVisitor(true, f))?;
       Ok(());
```